### DISCUSSION TITLE
No reaction after applying tag

### DISCUSSION THREAD

**Discussion by Matthew78917** [AloneLiberty/FlipperNested#3] [Q&A] [answered]
I successfully collected one key from `detect reader`, which allowed me to read one sector of the Mifare Classic 1K Card. I saved the card after it was done reading.

After, I went to the `Flipper (Mifare) Nested` app and clicked the `Nested attack`. It keeps on saying `Apply tag to the back`, which I did but the screen did not change. So I checked the logs on [https://lab.flipper.net/cli](https://lab.flipper.net/cli) `log debug` and I saw this:

![Screenshot 2023-05-04 045616](https://user-images.githubusercontent.com/54089351/236047747-c74b03d6-2a4b-46c3-953f-177215922843.png)

Red text appeared when I just clicked on the `Nested attack` button but didn't put the flipper onto the card yet. The blue text and `Timeout during data exchange` messages started appearing after I put the flipper onto the card. The screen on the flipper was still telling me to `Apply tag to the back`.

Apologies in advance for being a noob at this stuff, I'm not sure if I am doing it wrong. Does this have anything related to the detection of delay mentioned in todo?

**Comment by AloneLiberty** [comment]
Same goes for `Check found keys`?

**Reply by AloneLiberty** [reply]
Also `nfc apdu 6000` in CLI should return random bytes

**Reply by Matthew78917** [reply]
> Same goes for `Check found keys`?

yes, no response on flipper either. Still telling me to apply the tag. In the picture, I told it to stop nested attack and then I told it to check found keys
![image](https://user-images.githubusercontent.com/54089351/236064741-d03fc2b1-6b89-436b-92bb-d711065cb7c9.png)

**Reply by Matthew78917** [reply]
> Also `nfc apdu 6000` in CLI should return random bytes

This is what I get when running `nfc apdu 6000` I closed the app after the first error message
![image](https://user-images.githubusercontent.com/54089351/236065167-b7e902e6-f848-4890-bf38-106bc7b94583.png)

**Reply by AloneLiberty** [reply]
And flipper can read your card? Can you send here dump?

**Reply by Matthew78917** [reply]
Sector 11 Key A have, can read Sector 11. But I don't have access to any other sectors

```
Filetype: Flipper NFC device
Version: 3
# Nfc device type can be UID, Mifare Ultralight, Mifare Classic, Bank card
Device type: Mifare Classic
# UID, ATQA and SAK are common for all formats
UID: 04 34 79 9A 9D 71 80
ATQA: 00 44
SAK: 08
# Mifare Classic specific data
Mifare Classic type: 1K
Data format version: 2
# Mifare Classic blocks, '??' means unknown data
Block 0: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 1: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 2: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 3: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 4: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 5: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 6: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 7: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 8: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 9: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 10: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 11: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 12: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 13: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 14: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 15: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 16: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 17: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 18: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 19: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 20: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 21: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 22: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 23: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 24: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 25: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 26: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 27: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 28: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 29: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 30: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 31: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 32: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 33: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 34: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 35: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 36: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 37: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 38: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 39: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 40: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 41: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 42: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 43: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 44: FF 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Block 45: 00 8A 4C 41 DC 77 8B 00 00 18 53 68 14 0F 9A 43
Block 46: 25 79 33 33 33 33 33 33 30 33 44 20 00 20 00 00
Block 47: 40 D0 27 53 42 35 7F 07 88 20 ?? ?? ?? ?? ?? ??
Block 48: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 49: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 50: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 51: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 52: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 53: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 54: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 55: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 56: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 57: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 58: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 59: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 60: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 61: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 62: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
Block 63: ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? ??
```

**Comment by AloneLiberty** [comment]
Will `nfc apdu 602C` work?

**Reply by Matthew78917** [reply]
44BE response

![image](https://user-images.githubusercontent.com/54089351/236070340-74214c75-5ea1-4d73-b334-57f506be7c43.png)

**Reply by AloneLiberty** [reply]
Can you check from 00 to 0F (0123456789ABCDEF) and check when it will start working?

**Reply by Matthew78917** [reply]
![screencapture-lab-flipper-net-cli-2023-05-04-07_16_42](https://user-images.githubusercontent.com/54089351/236071616-8e9be013-ed20-43d7-98fa-8ca5f413e111.png)

 All failed to tx_rx

**Reply by AloneLiberty** [reply]
Don't forget about 60 at start, should be 6001, 6002, etc

**Reply by Matthew78917** [reply]
![screencapture-lab-flipper-net-cli-2023-05-04-07_21_36](https://user-images.githubusercontent.com/54089351/236072160-435c979c-b8ea-416c-862e-ecb9d70edd97.png)
 My bad, I forgot. Still all failed to tx_rx though

**Reply by AloneLiberty** [reply]
Hmm, 602B and 6030 aren't working?

**Reply by Matthew78917** [reply]
> Hmm, 602B and 6030 aren't working?

602B gives DD7A

6030 gives failed to tx_rx

Tried a few more, 6028 to 602F range works

**Reply by AloneLiberty** [reply]
> Tried a few more, 6028 to 602F range works

So seems they just bricked blocks for ...
You can try editing my code (search for 0x60, 0 and replace with 0x60, 0x2C), but I'm unsure that other code will able handle this. But at least you will get to calibration.

For my proper fix... idk when I will find time to dig into it, hopefully in May

**Reply by Matthew78917** [reply]
So just to confirm, I go to `FlipperNested-dev\lib\nested.c`, change on line 184 and then recompile a new .fap file right?

**Reply by AloneLiberty** [reply]
Yes

**Reply by Matthew78917** [reply]
> Yes 

ooooooo I think it might be working! I'm going to sleep now, will check on it when I wake up.

Thank you so much for your help! I'll mark it as resolved when I wake up if everything else works as well. Let me know if you want me to mark it as resolved right now

![image](https://user-images.githubusercontent.com/54089351/236077279-c18315a9-6679-4044-976a-ddced61d42a7.jpeg)

**Reply by Matthew78917** [reply]
In case you're wondering, `log debug` looks like this now 
![image](https://user-images.githubusercontent.com/54089351/236077848-4e76d82e-ba30-4396-97fa-7a3a8db2227f.png)

**Reply by AloneLiberty** [reply]
If states aren't going up - bad luck, should modify deeper

**Reply by AloneLiberty** [reply]
I will add temporary workaround tomorrow, should fix second (what you have now) issue
Can you create two issues (I'm bit too lazy)?
1. Feature request: set custom block for tag detection
2. Feature request: check if sector is alive in Nested attack

**Reply by Matthew78917** [reply]
> If states aren't going up - bad luck, should modify deeper 

So change to something like maybe 0x30 instead of 0x2C?

**Reply by Matthew78917** [reply]
> I will add temporary workaround tomorrow, should fix second (what you have now) issue Can you create two issues (I'm bit too lazy)?
> 
> 1. Feature request: set custom block for tag detection
> 2. Feature request: check if sector is alive in Nested attack

Of course, I'll make those two issues right now. Wait do you want want me to put them in issues tab or discussions tab?
![image](https://user-images.githubusercontent.com/54089351/236078614-fc227ba3-8860-4c65-a6ac-7d0575308ccb.png)

**Reply by AloneLiberty** [reply]
> > If states aren't going up - bad luck, should modify deeper
> 
> So change to something like maybe 0x30 instead of 0x2C?

Nothing that you can do probably, requires deep knowledge or VERY bad code
If you actually want to do it:
1. Find HardNested function (don't remember full name and from phone currently, it should be in worker.c)
2. There is check for sector is already has key, add check if sector doesn't match working one (in your case != 11)

**Reply by AloneLiberty** [reply]
> Of course, I'll make those two issues right now. Wait do you want want me to put them in issues tab or discussions tab?

Issues

**Reply by AloneLiberty** [reply]
https://github.com/AloneLiberty/FlipperNested/blob/fb317dfbfdff47b5f1847e4d76eda354af7b8180/mifare_nested_worker.c#L864
In this if add ` || sector != 11`

**Reply by AloneLiberty** [reply]
Also, your sector 10 is alive, might want to recover keys from it, just replace != 11 with != 10 (after you calculate keys!)

**Reply by Matthew78917** [reply]
![image](https://user-images.githubusercontent.com/54089351/236081172-c0ede508-3e79-4ce0-a40f-76b565f65782.jpeg)

Pog

**Comment by Matthew78917** [comment]
@AloneLiberty Thank you so much for all that you've done to help me! Today I have finally cracked my student ID card and read/write the data on it. I have been trying to do for 2 years on and off. Only got sector 11 and 10 though, even after changing the code to allow all sectors in the if statement, it still can't figure out the sector keys, but I'm assuming that's just a limitation or the school just had those sectors disabled maybe because you said they weren't alive? If you don't mind me asking, how did you know sector 10 was alive?

By the way, once 1.4 is out, do you want me to make a video tutorial on how to use this? I can't help out much with development because I suck at programming even though I'm a comp sci student, but I still want to help your project in any way possible.

**Reply by AloneLiberty** [reply]
You can write ACL (Access Bits) to card, that won't allow reading. That's what they did. Why I don't know, you can ask them yourself (they don't need them, but why exactly brick them down like that...)
I assumed sector 10 is alive, because you can do auth on 40 block, which is 10 sector.
I've made some progress with 1.4.0, if you have Telegram you can contact me and I will send you beta that checks for working sectors (still no ability to select checking block)

**Comment by AloneLiberty** [comment, marked as answer]
Fixed in 1.4.0
